urbit: 2015.09.26 -> 2016-06-02

brainrake opened the pull request:
###### Motivation for this change

urbit wasn't building because the source git repo changed and the relevant revision was missing
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Updated to latest revision in urbit git master, fixes build error caused by missing sources.
Updated description and website.
Fixed date/version format to ISO 8061.

joachifm commented:
Does urbit use submodules? If not, it is preferable to use a generated archive instead of a git checkout.

brainrape commented:
no submodules. updated to use github archive for the specific commit (there are no tags).

joachifm commented on pkgs/misc/urbit/default.nix:
```diff
@@ -1,14 +1,13 @@
-  src = fetchgit {
-    url = "https://github.com/urbit/urbit.git";
-    rev = "8c113559872e4a97bce3f3ee5b370ad9545c7459";
+  src = fetchzip {
```
If you prefer, `fetchFromGitHub` is a convenience wrapper which expands to what you've got here.

brainrape commented:
thanks for the tip

joachifm commented:
Looks good to me. Please squash the commits when you're happy to merge.

brainrake committed:
urbit: 2015.09.26 -> 2016-06-02

Updated to latest revision in urbit git master, fixes build error caused by missing sources.
Updated description and website.
Fixed date/version format to ISO 8061.

brainrape commented:
Squashed.
Cheers.

joachifm commented:
Thank you
